tests/expected/decode/test4.gz.stdout

```diff
@@ -13,24 +13,24 @@
 │           │   ├── blocks <- repeat-until-last (x -> x.final == 1) deflate.block :=
 │           │   │   ├── 0 <- deflate.block :=
 │           │   │   │   ├── final <- base.bit := 0
-│           │   │   │   ├── type <- { ... } := 2
+│           │   │   │   ├── type <- map (bits -> bits.1 << 1 | bits.0) (...) := 2
```

Comment:
It might be beneficial to have 'named lambdas' that display under a simple operation name, rather than force it to expand potentially verbose expressions into text. We could even add a threshold level for how many characters the expansion is allowed to take up before it falls back to the name, but that might be over-engineering an already somewhat extraneous feature...

Reply:
Yes we're a bit limited by the nature of a basic ASCII text dump here, but you could imagine that large values would be summarised or collapsed until explicitly expanded, in a more powerful UI.

Reply:
It might be overkill or a diversion, but the [tui crate](https://crates.io/crates/tui) looks pretty neat. It would be interesting, if we decided it was worth the effort, to see what we could do with a more feature-full interaction between the user and the output of the program.